eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/RegenerateUrlAliasesCommand.php

```diff
@@ -0,0 +1,1054 @@
+
+        $action = $input->getArgument('action');
+        $this->bulkCount = $input->getArgument('bulk-count');
+
+        if (!isset($this->actionSet[$action])) {
+            throw new RuntimeException("Action '{$action}' is not supported");
+        }
+
+        if ($action === 'full' || $action === 'backup-custom') {
+            $this->backupCustomLocationAliases();
+        }
+
+        if ($action === 'full' || $action === 'backup-global') {
+            $this->backupGlobalAliases();
+        }
+
+        if ($action === 'full' || $action === 'autogenerate') {
+            $this->generateLocationAliases();
+        }
+
+        if ($action === 'full' || $action === 'restore-custom') {
+            $this->restoreCustomLocationAliases();
+        }
+
+        if ($action === 'full' || $action === 'restore-global') {
+            $this->restoreGlobalAliases();
+        }
+    }
+
+    /**
+     * Checks that configured storage engine is Legacy Storage Engine.
+     */
+    protected function checkStorage()
+    {
+        $storageEngine = $this->getContainer()->get('ezpublish.api.storage_engine');
+
+        if (!$storageEngine instanceof LegacyStorageEngine) {
+            throw new RuntimeException(
+                'Expected to find Legacy Storage Engine but found something else.'
+            );
+        }
+    }
+
+    /**
+     * Prepares dependencies used by the command.
+     *
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     */
+    protected function prepareDependencies(OutputInterface $output)
+    {
+        /** @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler $databaseHandler */
+        $databaseHandler = $this->getContainer()->get('ezpublish.connection');
+        /** @var \eZ\Publish\Core\Persistence\Legacy\Content\UrlAlias\Gateway $gateway */
+        $gateway = $this->getContainer()->get('ezpublish.persistence.legacy.url_alias.gateway');
+        /** @var \eZ\Publish\SPI\Persistence\Handler $persistenceHandler */
+        $persistenceHandler = $this->getContainer()->get('ezpublish.api.persistence_handler');
+        /** @var \eZ\Publish\Core\Repository\Repository $innerRepository */
+        $innerRepository = $this->getContainer()->get('ezpublish.api.inner_repository');
+        /** @var \eZ\Publish\API\Repository\Repository $repository */
+        $repository = $this->getContainer()->get('ezpublish.api.repository');
+
+        $administratorUser = $repository->getUserService()->loadUser(14);
+        $repository->getPermissionResolver()->setCurrentUserReference($administratorUser);
+
+        $this->contentService = $repository->getContentService();
+        $this->nameSchemaResolver = $innerRepository->getNameSchemaService();
+        $this->urlAliasHandler = $persistenceHandler->urlAliasHandler();
+        $this->urlAliasGateway = $gateway;
+        $this->connection = $databaseHandler->getConnection();
+        $this->output = $output;
+    }
```

Comment:
Great, changed in https://github.com/ezsystems/ezpublish-kernel/pull/1781/commits/69ca86353751dec06a4e26dea163643598c62e31